### ISSUE TITLE
Document that Z is extracted to either int or bigint depending on Coq version

### ISSUE THREAD

**Issue by anton-trunov** [QuickChick/QuickChick#294]
I was using Coq 8.14 and the mapping from `Z` to `int` and it was a bit of surprise to get a compilation error on Coq 8.15. It would be great to have this documented somewhere; perhaps in the release notes?

I guess my best option to keep compatibility with both Coq versions is to switch to bigints for 8.14?

For easier reference: related PR is #227.

**Comment by liyishuai** [comment]
Yes it is worth extracting Z to big integer everywhere. Maybe worth moving `ExtrOcamlZBigInt` to a package that supports multiple versions of Coq.
